snapcraft/plugins/go.py

```diff
@@ -101,7 +102,13 @@ def get_pull_properties(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
-        self.build_packages.append('golang-go')
+
+        # Do not try to install golang if it is already available in our PATH.
+        try:
+            subprocess.check_output(['go', 'version'])
+        except OSError:
+            self.build_packages.append('golang-go')
```

Comment:
This will effectively "hide" the Go being used from the collected artifacts, won't it? Since there's no package with a specific version to track.
@elopio Do we have a story here yet?